Mathlib/Algebra/Category/ModuleCat/ChangeOfRings.lean

```diff
@@ -445,51 +434,42 @@ variable {R : Type u₁} {S : Type u₂} [Ring R] [Ring S] (f : R →+* S)
 /-- Given `R`-module X and `S`-module Y, any `g : (restrictScalars f).obj Y ⟶ X`
 corresponds to `Y ⟶ (coextendScalars f).obj X` by sending `y ↦ (s ↦ g (s • y))`
 -/
-@[simps apply_apply]
+@[simps hom_apply_apply]
 def HomEquiv.fromRestriction {X : ModuleCat R} {Y : ModuleCat S}
-    (g : (restrictScalars f).obj Y ⟶ X) : Y ⟶ (coextendScalars f).obj X where
-  toFun := fun y : Y =>
-    { toFun := fun s : S => g <| (s • y : Y)
-      map_add' := fun s1 s2 : S => by simp only [add_smul]; rw [LinearMap.map_add]
-      map_smul' := fun r (s : S) => by
-        -- Porting note: dsimp clears out some rw's but less eager to apply others with Lean 4
-        dsimp
-        rw [← g.map_smul]
-        erw [smul_eq_mul, mul_smul]
-        rfl}
-  map_add' := fun y1 y2 : Y =>
-    LinearMap.ext fun s : S => by
-      -- Porting note: double dsimp seems odd
-      dsimp only [id_eq, eq_mpr_eq_cast, AddHom.toFun_eq_coe, AddHom.coe_mk, RingHom.id_apply,
-        RingHom.toMonoidHom_eq_coe, OneHom.toFun_eq_coe,
-        MonoidHom.toOneHom_coe, MonoidHom.coe_coe, ZeroHom.coe_mk, smul_eq_mul, cast_eq,
-        LinearMap.coe_mk]
-      rw [LinearMap.add_apply, LinearMap.coe_mk, LinearMap.coe_mk]
-      dsimp only [AddHom.coe_mk]
-      rw [smul_add, map_add]
-  map_smul' := fun (s : S) (y : Y) => LinearMap.ext fun t : S => by
-      -- Porting note: used to be simp [mul_smul]
-      simp only [LinearMap.coe_mk, AddHom.coe_mk, RingHom.id_apply]
-      rw [ModuleCat.CoextendScalars.smul_apply', LinearMap.coe_mk]
-      dsimp
-      rw [mul_smul]
+    (g : (restrictScalars f).obj Y ⟶ X) : Y ⟶ (coextendScalars f).obj X :=
+  asHom
```

Comment:
The `@[simps]` lemma generated here has the wrong type, but if we type-ascript `asHom` to give it the right type, then the proofs in this definition breaks.